partiql-ast/src/ast.rs

```diff
@@ -967,20 +915,22 @@ pub enum Type {
 
 #[derive(Clone, Debug, PartialEq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct CharacterType {
-    pub length: Option<LongPrimitive>,
+pub enum CustomTypeParam {
+    Lit(Lit),
+    Type(Type),
```

Comment:
Curious: do we have a sample for a type that accepts a type as its parameter now? no objection for having it btw.